Makefile

```diff
@@ -140,7 +140,15 @@ init_upstream:
 			(cd upstream && git submodule update --init && git remote add source git@github.com:hashicorp/terraform-provider-aws.git) ; \
 		fi; \
 
-patch_upstream: init_upstream
+export_upstream_patches: init_upstream
+ifeq ($(shell cd upstream && git rev-parse --is-shallow-repository), false)
+	find upstream-patches -type f -delete
+	cd upstream && \
+		LAST_TAG=$$(git describe --abbrev=0 --tags) && \
+		git format-patch  -o ../upstream-patches --minimal --no-signature HEAD...$${LAST_TAG}
```

Comment:
Generally, `HEAD` could be anything. You're relying on the assumption that it's the latest `patched-...` branch as per the upgrade playbook? Can we make that safer?

Reply:
The challenge here is that if we've just checked out the submodule directly we won't be on a branch, but a detached head - pointing directly to the commit hash in the parent repo. I think the risk is low as you'd just see a completely different set of patches exported which should raise questions in the review.